kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroDataConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.s3.format.avro;
+
+import io.confluent.connect.storage.common.ComposableConfig;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.Map;
+
+import static io.confluent.connect.avro.AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG;
+import static io.confluent.connect.avro.AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_DEFAULT;
+import static io.confluent.connect.avro.AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_DOC;
+import static io.confluent.connect.avro.AvroDataConfig.CONNECT_META_DATA_CONFIG;
+import static io.confluent.connect.avro.AvroDataConfig.CONNECT_META_DATA_DEFAULT;
+import static io.confluent.connect.avro.AvroDataConfig.CONNECT_META_DATA_DOC;
+import static io.confluent.connect.avro.AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG;
+import static io.confluent.connect.avro.AvroDataConfig.SCHEMAS_CACHE_SIZE_DEFAULT;
+import static io.confluent.connect.avro.AvroDataConfig.SCHEMAS_CACHE_SIZE_DOC;
+
+/**
+ * Adapts {@link io.confluent.connect.avro.AvroDataConfig} which uses {@link io.confluent.common.config.AbstractConfig} which is
+ * incompatible with {@link org.apache.kafka.common.config.AbstractConfig}.
+ */
+public class AvroDataConfig extends AbstractConfig implements ComposableConfig {
```

Comment:
It's indeed unfortunate that we can't inherit from `AvroDataConfig` in schema-registry. It would be more bullet proof to future changes/additions to this config class (that's how we missed passing this functionality here in the first place. It was added in SR but not in the sink connectors). 

Still, since this is going to be used by the hdfs connector as well, I'll take this part and issue a PR in kafka-connect-storage-common instead.

Reply:
I'll also call it `ComposableAvroDataConfig`. Maybe not a great name, but will help with conflicts with the original type of `AvroDataConfig` which we'll need to use when accessing the static members of this class.